### PR TITLE
DOC: update DatetimeTZDtype unit limitation

### LINKED ISSUE #58212: DOC: DatetimeTZDtype unit limitation is not updated
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/dev/reference/api/pandas.DatetimeTZDtype.html

> unit: str, default “ns”
> The precision of the datetime data. Currently limited to "ns".****

### Documentation problem

The document says `Currently limited to "ns".`,  but actually `DatetimeTZDtype` seems to support non-nano since https://github.com/pandas-dev/pandas/pull/47120 

### Suggested fix for documentation

It should be like `Currently limited to "s", "ms", "us", "ns".`

### PR DESCRIPTION
**Pull request by yokomotod**
- [x] closes #58212
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/dtypes/dtypes.py [hunk 1]
```diff
@@ -699,7 +699,7 @@ class DatetimeTZDtype(PandasExtensionDtype):
     ----------
     unit : str, default "ns"
         The precision of the datetime data. Currently limited
```
Comment (Aloqeely) [inline pandas/core/dtypes/dtypes.py:701]: I think we should consider deleting "Currently limited" since there are 4 possible options now.

Comment (yokomotod) [reply pandas/core/dtypes/dtypes.py:701]: @Aloqeely thank you for your review.

How about

```
The precision of the datetime data. Only supports
``"s"``, ``"ms"``, ``"us"``, ``"ns"``.
```

as same as error message?

```py
if unit not in ["s", "ms", "us", "ns"]:
    raise ValueError("DatetimeTZDtype only supports s, ms, us, ns units")
```

Comment (Aloqeely) [reply pandas/core/dtypes/dtypes.py:701]: Yes sure, `Only supports` and `Valid options are` both sound good to me.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @yokomotod

**Pull request merged**